File: ruby/FuzzyPlayer.rb
# encoding: UTF-8

require_relative 'Player' 
require_relative 'Dice'

module Irrgarten
  
  class FuzzyPlayer < Player

    def initialize(other)
      copy(other)
    end

    def move(direction, valid_moves)
      Dice.next_setp(super, valid_moves, intelligence)
    end

    def attack
      self.sum_weapons + Dice.intensity(@strength)
    end

    def to_s
      string = "Fuzzy"
      string += super
      return string
    end

    protected
    def defensive_energy
      return self.sum_shields + Dice.intensity(@intelligence)
    end

  end # class

end # module

File: java/src/irrgarten/FuzzyPlayer.java
package irrgarten;

import java.util.ArrayList;
public class FuzzyPlayer extends Player {
   
    //Constructor:
    public FuzzyPlayer(Player other) {
        super(other);
    }

    @Override
    public Directions move(Directions direction, ArrayList<Directions> validMoves) {
        Directions dir = super.move(direction, validMoves);
        return Dice.nextStep(dir, validMoves, getIntelligence());

    }
    
    @Override
    public float attack() {
        return (sumWeapons() + Dice.intensity(getStrength()));
    }

    @Override
    protected float defensiveEnergy() {
        return (sumShields() + Dice.intensity(getIntelligence()));
        
    }

    @Override
    public String toString(){
        return ("Fuzzy"+super.toString());
    }
    
}
